scio-neo4j/src/main/scala/com/spotify/scio/neo4j/Neo4jIO.scala

```diff
@@ -81,12 +87,24 @@ case class Neo4jCypher[T: Coder](readOptions: Neo4jReadOptions[T]) extends Neo4j
   override def tap(read: ReadP): Tap[Nothing] = EmptyTap
 }
 
-final case class Neo4jWrite[T](writeOptions: Neo4jWriteOptions[T]) extends Neo4jIO[T] {
+final case class Neo4jWriteUnwind[T: ParametersBuilder](neo4jOptions: Neo4jOptions, cypher: String)
+    extends Neo4jIO[T] {
   override type ReadP = Nothing
-  override type WriteP = Unit
+  override type WriteP = Neo4jIO.WriteParam
   override val tapT: TapT.Aux[T, Nothing] = EmptyTapOf[T]
 
-  override def testId: String = s"Neo4jIO(${Neo4jIO.neo4jIoId(writeOptions)})"
+  private[neo4j] val unwindMapName: String = cypher match {
+    case Neo4jIO.UnwindParameterRegex(name) => name
+    case _ =>
+      throw new IllegalArgumentException(
+        s"""Expected unwind cypher with parameter but got:
+         |$cypher
+         |See: https://neo4j.com/docs/cypher-manual/current/clauses/unwind/#unwind-creating-nodes-from-a-list-parameter
+         |""".stripMargin
+      )
+  }
```

Comment:
Trying to be smart here: we can skip passing `unwindMapName` manually and extract if from the query.
Since we only support `writeUnwind` I think this is safe